src/controllers/llmo/llmo-onboarding.js

```diff
@@ -0,0 +1,373 @@
+ * @param {object} context - The request context
+ * @param {object} slackContext - Slack context (optional, for Slack operations)
+ * @returns {Promise<object>} The organization object
+ */
+export async function createOrFindOrganization(imsOrgId, context, slackContext = null) {
+  const { dataAccess, log } = context;
+  const { Organization } = dataAccess;
+  const { say } = slackContext || { say: () => {} };
+
+  // Check if organization already exists
+  let organization = await Organization.findByImsOrgId(imsOrgId);
+
+  if (organization) {
+    log.debug(`Found existing organization for IMS Org ID: ${imsOrgId}`);
+    return organization;
+  }
+
+  // Create new organization
+  log.info(`Creating new organization for IMS Org ID: ${imsOrgId}`);
+  await say(`Creating organization for IMS Org ID: ${imsOrgId}`);
+
+  organization = await Organization.create({
+    name: `Organization ${imsOrgId}`,
+    imsOrgId,
+  });
+
+  log.info(`Created organization ${organization.getId()} for IMS Org ID: ${imsOrgId}`);
+  return organization;
+}
+
+/**
+ * Creates entitlement and enrollment for LLMO.
+ * @param {object} site - The site object
+ * @param {object} context - The request context
+ * @returns {Promise<void>}
+ */
+export async function createEntitlementAndEnrollment(site, context) {
+  const { log } = context;
+
+  try {
+    const tierClient = await TierClient.createForSite(context, site, LLMO_PRODUCT_CODE);
+    const { entitlement, siteEnrollment } = await tierClient.createEntitlement(LLMO_TIER);
+    log.info(`Successfully ensured LLMO access for site ${site.getId()} via entitlement ${entitlement.getId()} and enrollment ${siteEnrollment.getId()}`);
+
+    return {
+      entitlement,
+      enrollment: siteEnrollment,
+    };
+  } catch (error) {
+    log.info(`Ensuring LLMO entitlement and enrollment failed: ${error.message}`);
+    throw error;
+  }
+}
+
+/**
+ * Complete LLMO onboarding process.
+ * @param {object} params - Onboarding parameters
+ * @param {string} params.domain - The domain name
+ * @param {string} params.brandName - The brand name
+ * @param {string} params.imsOrgId - The IMS Organization ID
+ * @param {object} context - The request context
+ * @param {object} slackContext - Slack context (optional, for Slack operations)
+ * @returns {Promise<object>} Onboarding result
+ */
+export async function performLlmoOnboarding(params, context) {
+  const { domain, brandName, imsOrgId } = params;
+  const { dataAccess, env, log } = context;
+  const { Site } = dataAccess;
+
+  // Construct base URL and data folder name
+  const baseURL = domain.startsWith('http') ? domain : `https://${domain}`;
+  const dataFolder = generateDataFolder(domain, env.ENV);
+
+  log.info(`Starting LLMO onboarding for IMS org ${imsOrgId}, domain ${domain}, brand ${brandName}`);
+
+  // Create or find organization
+  const organization = await createOrFindOrganization(imsOrgId, context);
+
+  // Create site
+  const site = await Site.create({
```

Comment:
site might have exists in spacecat, no? I think we need `createOrFindSite` here, just like organization

Reply:
also, if the site exists, we need to check if it belongs to the organization (or default org) to make sure we are not assigning the site to wrong org